src/main/java/mate/academy/springboot/datajpa/repository/ProductRepository.java

```diff
@@ -0,0 +1,14 @@
+package mate.academy.springboot.datajpa.repository;
+
+import java.math.BigDecimal;
+import java.util.List;
+import mate.academy.springboot.datajpa.model.Product;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface ProductRepository extends JpaRepository<Product, Long> {
+    List<Product> findAllByPriceBetween(BigDecimal from, BigDecimal to);
+
+    List<Product> findProductsByCategoryNameIn(List<String> categories);
```

Comment:
Redundant. You should create this method in ProductServiceImpl, not in Repository, and use `Map` in the constructor of the method

Reply:
it's not true. Better to use Spring Data to build queries and this way is right